bin/kano-world-hook

```diff
@@ -28,46 +36,56 @@ from kano.logging import logger
 
 # TODO: Put kano-dialogs where appropriate
 
-if len(sys.argv) < 2:
-    msg = "{} run with no arguments".format(sys.argv[0])
-    logger.error(msg)
-    sys.exit(0)
 
-# Parse the URL
-try:
-    url_data = sys.argv[1].split(":")
-    if len(url_data) < 3:
-        print '{} url data is too short: {}'.format(sys.argv[0], sys.argv[1])
+def main(scheme_url, **args):
+    # Parse the URL
+    try:
+        url_data = scheme_url.split(":")
+        if len(url_data) < 3:
+            msg = '{} url data is too short: {}'.format(sys.argv[0], scheme_url)
+            logger.error(msg)
+            sys.exit(0)
+    except:
+        msg = '{} passed an invalid url: {}'.format(sys.argv[0], scheme_url)
         logger.error(msg)
         sys.exit(0)
-except:
-    print '{} passed an invalid url: {}'.format(sys.argv[0], sys.argv[1])
-    sys.exit(0)
 
-try:
-    scheme = url_data.pop(0)
-    assert(scheme=='kano')
-except:
-    msg = "Could not recognise the {} scheme".format(scheme)
-    logger.error(msg)
-    sys.exit(0)
+    # First element is the scheme specifier - should be `kano`
+    try:
+        scheme = url_data.pop(0)
+        assert(scheme=='kano')
+    except:
+        msg = "Could not recognise the {} scheme".format(scheme)
+        logger.error(msg)
+        sys.exit(0)
 
-try:
-    module_name = "kano_world.hooks.{}".format(url_data[0])
-    hook_module = importlib.import_module(module_name)
-except:
-    msg = "The requested '{}' hook doesn't exist".format(url_data[0])
-    logger.error(msg)
-    sys.exit(0)
+    # Second element is the scheme action (e.g. share, launch, etc)
+    try:
+        module_name = "kano_world.hooks.{}".format(url_data[0])
+        hook_module = importlib.import_module(module_name)
+    except:
+        msg = "The requested '{}' hook doesn't exist".format(url_data[0])
+        logger.error(msg)
+        sys.exit(0)
+
+    # Remaining elements are parameters for scheme action
+    try:
+        result = hook_module.run(url_data[1:])
+    except Exception as e:
+        msg = "The execution of the {} hook failed ({})".format(
+            sys.argv[1], str(e)
+        )
+        logger.error(msg)
+        sys.exit(0)
+
+    if hasattr(hook_module, "launch"):
+        hook_module.launch(result, background=args.get('--background', False))
 
-try:
-    result = hook_module.run(url_data[1:])
-except Exception as e:
-    msg = "The execution of the {} hook failed ({})".format(sys.argv[1], str(e))
-    logger.error(msg)
     sys.exit(0)
 
-if hasattr(hook_module, "launch"):
-    hook_module.launch(result)
 
-sys.exit(0)
+if __name__ == '__main__':
+    ARGS = docopt.docopt(__doc__)
+    SCHEME_URL = ARGS.get('<kano-scheme-url>')
+
+    main(SCHEME_URL, **ARGS)
```

Comment:
Not really important, but another way of doing it is to:
```
sys.exit(main(SCHEME_URL, **ARGS))
```
and use `return RC_x` inside the `main` function.